src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -208,6 +210,22 @@ export class KeyStrokeHandler {
             }
         }
     }
+
+    /**
+     * Will try run given labmda function while keyStrokeHandler is turned off, then turn keyStrokeHandler on after execution no matter the execution succeed or not
```

Comment:
Can you explain how this works? Is there any case when this._enabled will be left as false forever?

Reply:
> Can you explain how this works? Is there any case when this._enabled will be left as false forever?
nope, finally will always execute so even exception thrown, it will be turned on

Reply:
>When detecting document change event, if the event.text is the last recommendation, then it must come from the acceptance behavior. Just do a string comparison.


make sense let me try this